### PR TITLE
avoiding allocations in ConditionalNodes

### PR DESCRIPTION
**Pull request by dfa1**
Alternative fix for #2119 

Master:
```
Benchmark                                    Mode  Cnt   Score   Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  42.681 ± 1.912  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15  23.026 ± 0.621  ms/op
```

This branch:
```
Benchmark                                    Mode  Cnt   Score   Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  46.075 ± 2.372  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15  21.021 ± 0.602  ms/op
```

### CODE REVIEW

File: src/main/java/graphql/execution/ConditionalNodes.java [hunk 1]
```diff
@@ -28,19 +25,23 @@ public boolean shouldInclude(Map<String, Object> variables, List<Directive> dire
...
             Object flag = argumentValues.get("if");
             Assert.assertTrue(flag instanceof Boolean, () -> String.format("The '%s' directive MUST have a value for the 'if' argument", directiveName));
             return (Boolean) flag;
         }
         return defaultValue;
     }
 
-    private List<Directive> getDirectiveByName(List<Directive> directives, String name) {
-        return NodeUtil.allDirectivesByName(directives).getOrDefault(name, emptyList());
+    private Directive findDirectiveByName(List<Directive> directives, String name) {
```
Comment (dugenkui03) [inline src/main/java/graphql/execution/ConditionalNodes.java:38]: Move the logic of `findDirectiveByName` to `NodeUtil`, and replace all `getXXByName` with `getNodeByName(List<NamedNode> list, String name)` will be better.

Comment (dfa1) [reply src/main/java/graphql/execution/ConditionalNodes.java]: done

**Comment by andimarek** [comment]
@dfa1 can you actually confirm that it fixes your hotspot?

**Comment by dfa1** [comment]
it would be much more easy to merge and test from maven central. 

By the way if you need a bugfix release urgently, I could wait after that

**Comment by dfa1** [comment]
@andimarek @bbakerman  would it be possible to merge this PR as well?

**Comment by bbakerman** [comment]
I am happy for this to be merged

**Comment by dfa1** [comment]
thank you very much!

**Pull request merged**